src/uthenticode-config.cmake

```diff
@@ -0,0 +1,6 @@
+include(CMakeFindDependencyMacro)
+
+find_dependency(OpenSSL 1.1 COMPONENTS Crypto)
```

Comment:
Just a question: is `REQUIRED` not necessary/not preferable here? This is outside of what I know about CMake 😄

Reply:
Good question, and if I'm interpreting the docs[1] correctly, the `find_dependency` is used when `uthenticode` already `REQUIRED` OpenSSL, and so projects that require `uthenticode` must also find OpenSSL.

If `uthenticode` had an optional dependency `LibA` that it was configured to not use (when installed), then there should not be a `find_dependency(LibA)`.

I believe this `-config` file should represent what dependencies are required to find and use `uthenticode` when you write `find_package(uthenticode REQUIRED)`. On the other hand, if you use `find_package(uthenticode QUIET)`, CMake will forward that `QUIET` option to the `find_dependency` listed in `uthenticode-config.cmake` and not fail if it can't find the dependencies.

[1] https://cmake.org/cmake/help/v3.18/module/CMakeFindDependencyMacro.html#module:CMakeFindDependencyMacro

Reply:
Gotcha! That makes sense, thanks for the detailed explanation.

Reply:
The "Creating a package configuration file" section also has this quote

> The `find_dependency` macro from the CMakeFindDependencyMacro helps with this by propagating whether the package is `REQUIRED`, or `QUIET` etc. All `REQUIRED` dependencies of a package should be found in the Config.cmake file:

https://cmake.org/cmake/help/v3.18/manual/cmake-packages.7.html#creating-a-package-configuration-file